lib/puppet/provider/cpdb_update/cpdb_update.rb

```diff
@@ -0,0 +1,46 @@
+Puppet::Type.type(:cpdb_update).provide(:cpdb_update) do
+
+  commands :cpdb => '/usr/share/candlepin/cpdb'
+  commands :rpm => 'rpm'
+
+  def create
+    migrate_database
+    update_version_file
+  end
+
+  def exists?
+    Gem::Version.new(previous_candlepin_version) == Gem::Version.new(candlepin_rpm_version)
```

Comment:
I'd handle nil better:
```
[2] pry(main)> Gem::Version.new(nil)
nil versions are discouraged and will be deprecated in Rubygems 4
=> Gem::Version.new("")
```

In either case, if one of the two is nil, you also want to return false.